app/controllers/spree/credit_cards_controller.rb

```diff
@@ -74,6 +81,15 @@ def create_customer(token)
       Stripe::Customer.create(email: spree_current_user.email, source: token)
     end
 
+    def verify(card)
+      options = { metadata: { adding_card: true } }
+      Spree::Gateway::StripeSCA.new.verify(card, options)
```

Comment:
In terms of code structure, if you dont use an existing payment method and do this call to new StripeSCA. I think this is breaking the existing design and adding a very different way to access this code in the gateway, that is an indication to me this code should be in a different class. So, if we go this way, I think this verify call should be in a separate service, not inside the StripeSCA gateway.